File: test/standalone/java/SocialNetwork.java
package generated;

import grakn.client.GraknClient;
import graql.lang.Graql;
import graql.lang.query.GraqlQuery;
import grakn.core.rule.GraknTestServer;


import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Paths;

import org.junit.*;

public class TestStandaloneSocialNetwork {

    @ClassRule
    public static final GraknTestServer server = new GraknTestServer(
            Paths.get("test/grakn-test-server/conf/grakn.properties"),
            Paths.get("test/grakn-test-server/conf/cassandra-embedded.yaml")
    );

    @BeforeClass
    public static void loadSocialNetwork() {
        GraknClient client = new GraknClient(server.grpcUri().toString());
        GraknClient.Session session = client.session("social_network");
        GraknClient.Transaction transaction = session.transaction().write();

        try {
            byte[] encoded = Files.readAllBytes(Paths.get("files/social-network/schema.gql"));
            String query = new String(encoded, StandardCharsets.UTF_8);
            transaction.execute((GraqlQuery) Graql.parse(query));
            transaction.commit();
            session.close();
            System.out.println("Loaded the social_network schema");
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    @Test
    public void testSocialNetworkQuickstartQuery() {
        String[] args = { server.grpcUri().toString() };
        SocialNetworkQuickstartQuery.main(args);
    }

    @Test
    public void testGraknQuickstartA() {
        String[] args = { server.grpcUri().toString() };
        GraknQuickstartA.main(args);
    }

    @Test
    public void testGraknQuickstartB() {
        String[] args = { server.grpcUri().toString() };
        GraknQuickstartB.main(args);
    }

    @Test
    public void testGraknQuickstartC() {
        String[] args = { server.grpcUri().toString() };
        GraknQuickstartC.main(args);
    }

    @AfterClass
    public static void cleanSocialNetwork() {
        GraknClient client = new GraknClient(server.grpcUri().toString());
        client.keyspaces().delete("social_network");
        System.out.println("Deleted the social_network keyspace");
    }
}

File: test/example/python/social_network.py
import grakn
import unittest


class TestStandaloneSocialNetwork(unittest.TestCase):

    @classmethod
    def setUpClass(cls):
        with open('files/social-network/schema.gql', 'r') as schema:
            define_query = schema.read()

            client = grakn.Grakn(uri="localhost:48555")
            with client.session("social_network") as session:
                with session.transaction(grakn.TxType.WRITE) as transaction:
                    transaction.query(define_query)
                    transaction.commit()
                    print("Loaded the social_network schema")

    def test_social_network_quickstart_query(self):
        import social_network_quickstart_query

    def test_social_network_python_client_a(self):
        import social_network_python_client_a

    def test_social_network_python_client_b(self):
        import social_network_python_client_b

    def test_social_network_python_client_c(self):
        import social_network_python_client_c

    def test_social_network_python_client_d(self):
        import social_network_python_client_d

    @classmethod
    def tearDownClass(cls):
        client = grakn.Grakn(uri="localhost:48555")
        client.keyspaces().delete("social_network")
        print("Deleted the social_network keyspace")


if __name__ == '__main__':
    unittest.main()
